pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1274,3 +1274,18 @@ def func(ser):
 
     expected = DataFrame([[1.0]], index=[1])
     tm.assert_frame_equal(res, expected)
+
+
+@pytest.mark.parametrize("dtype", ["int64", "int32", "float64", "float32"])
+def test_groupby_agg_precision(dtype):
+    # GH33234
+    df = DataFrame(
+        {"key1": ["a"], "key2": ["b"], "key3": [1583715738627261039]}
+    ).astype({"key3": dtype})
+    df1 = df.groupby(["key1"]).agg(lambda x: x)
+    df2 = df.groupby(["key1", "key2"]).agg(lambda x: x)
+    expected = df.iloc[0]["key3"]
+    result1 = df1.iloc[0]["key3"]
+    result2 = df2.iloc[0]["key3"]
+    assert result1 == expected
```

Comment:
Instead, please construct a `result` and `expected` dataframe and use `tm.assert_frame_equal`

```
result = df.groupby(["key1", "key2"]).agg(lambda x: x)
expected = pd.DataFrame(...)
tm.assert_frame_equal(result, expected)
```

Reply:
Hello--thank you for your notes.

When I run the below code, these two data frames evaluate as equal when using `assert_frame_equal`. 

```
df1 = pd.DataFrame(
        {"key1": ["a"], "key2": ["b"], "key3": [1583715738627261039]}
    ).astype({"key3": 'int64'})

df1 = df1.groupby(["key1"]).agg(lambda x: x).reset_index()

df2 = pd.DataFrame(
        {"key1": ["a"], "key2": ["b"], "key3": [1583715738627260928]}
    ).astype({"key3": 'int64'})

df2 = df2.groupby(["key1", "key2"]).agg(lambda x: x).reset_index()

pd.testing.assert_frame_equal(df1, df2)
```

Reply:
Oh, but I didn't try this code from the source. Let me check that, and I'll update this thread.

Reply:
Yes, everything is fine. I'll submit this update. Thanks!